tools/monitoring/criersub/subscriber.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package criersub
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// ReportMessage is a message structure used to pass a prowjob status to Pub/Sub topic.
+// TODO(yt3liu): Use the ReportMessage structure in "k8s.io/test-infra/prow/pubsub/reporter"
+type ReportMessage struct {
+	Project string `json:"project"`
+	Topic   string `json:"topic"`
+	RunID   string `json:"runid"`
+	Status  string `json:"status"`
+	URL     string `json:"url"`
+	GCSPath string `json:"gcs_path"`
+}
+
+// SubscriberClient provide methods to run SubscriptionOperation
+// It implements all methods in SubscriberOperation
+type SubscriberClient struct {
+	SubscriberOperation
```

Comment:
I'm not sure if you need this field if it's just meant to implement interface

Reply:
Maybe I am missing something, but I am not able to define `ReceiveMessageAckAll` without it.

Reply:
Interesting, I'm not seeing anything special about ReceiveMessageAckAll